src/common/components/mapView/MapView.tsx

```diff
@@ -78,16 +101,28 @@ function MapView({ items = [], center, zoom }: Props) {
           maxClusterRadius={60}
           animate={false}
         >
-          {items?.map((item) => {
+          {items.map((item) => {
             if (!item.location) {
               return null;
             }
 
+            // used to automatically open popup for focused map item
+            // react-leaflet doesn't seems to provide better wway to do this
+            const handleOpenPopup = (marker) => {
+              const isFocusedItem = focusedItemId && item.id === focusedItemId;
+              if (isFocusedItem) {
+                // doesn't work without setTimeout. There seems to be some asynchronity
+                // in leaflet libary when rendering markers.
+                setTimeout(() => marker?.openPopup());
+              }
+            };
```

Comment:
This is likely something the React API doesn't provide for and you would need to integrate into Leaflet directly. Maybe it'd be possible to do with `Map.load` for instance. You can find the map instance with `useMap`. Just context, I don't think you need to change the current approach, it works for me.